tests/unit/rag/test_query_generator.py

```diff
@@ -80,29 +80,25 @@ def test_create_chain(self, mock_model, mock_llm):
             (
                 "How to deploy function?",
                 None,
-                None,
+                Queries(queries=["How to deploy function?"]),
                 Exception("Error generating queries"),
             ),
         ],
     )
     @pytest.mark.asyncio
     async def test_agenerate_queries(
-        self, mock_model, mock_llm, query, chain_output, expected_output, expected_error
+        self, mock_model, mock_llm, query, chain_output, expected_output, error
     ):
         """Test agenerate_queries method."""
         mock_chain = AsyncMock()
         with patch.object(QueryGenerator, "_create_chain", return_value=mock_chain):
             generator = QueryGenerator(mock_model)
             # When/Then
-            if expected_error:
-                mock_chain.ainvoke.side_effect = expected_error
-
-                with pytest.raises(type(expected_error)) as exc_info:
-                    await generator.agenerate_queries(query)
-                assert str(exc_info.value) == str(expected_error)
+            if error:
+                mock_chain.ainvoke.side_effect = error
             else:
                 mock_chain.ainvoke.return_value = chain_output
```

Comment:
Could this now conceal errors that we might want to detect?

Reply:
yes, it writes the error in the log with stacktrace. However, it still returns the original query as a fallback so that RAG can still search for documents with the original query.

Reply:
If you say yes, the changed test will conceal errors will that we actually would want to detect here, why would we do this?